linux/shim/service.go

```diff
@@ -101,7 +101,6 @@ func (s *Service) Exec(ctx context.Context, r *shimapi.ExecRequest) (*shimapi.Ex
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	s.execID++
-	reaper.Default.Lock()
```

Comment:
The point is that we hold the reaper lock so that the race that you introduce here isn't an issue.